### ISSUE TITLE
Cannot override config sections in environment spack.yaml

### ISSUE THREAD

**Issue by scheibelp** [spack/spack#10550]
The `spack.yaml` environment specification allows users to add their own config (https://spack.readthedocs.io/en/latest/tutorial_environments.html#configuration) according to the familiar schemas used outside of environments (https://spack.readthedocs.io/en/latest/configuration.html). 

However, while non-environment configs allow you to override lower-precedence scopes using `::` like

```
config::
  install_tree: /some/other/directory
```

(see https://spack.readthedocs.io/en/latest/configuration.html#overriding-entire-sections)

The same cannot be done within an environment. This can be problematic if a user wants the environment configuration to ignore the configuration stored in `~/.spack`.

### Steps to reproduce the issue

```console
$ (edit test-spack.yaml)
$ spack env create env1 test-spack.yaml
==> Error: spack-test.yaml:2: Additional properties not allowed ('config:' was unexpected)
```

Note that for individual package overrides, you can use the `::` syntax,

```
spack:
  specs:
    - python@3.6.1
  packages:
    python::
      variants: +optimizations
```

but you cannot override entire sections

**Comment by adrienbernede** [comment]
@scheibelp What is preventing this issue to be closed? It seems it was solved by #11602, wasn’t it?

**Comment by scheibelp** [comment]
Hi @adrienbernede yes this issue is now fixed - thanks for pointing that out!

The issue was that I created two PRs for this: the first one (which explicitly claimed to close this) was closed rather than merged; the second one (#11602) did not explicitly claim to fix this (although that's what it does), so when it was merged this issue was not automatically closed by github.

**Comment by adrienbernede** [comment]
Greg pointed this issue to me, and I had to follow the same strange path to realize it was solved.
Glad I could help.
